cli/azd/pkg/azsdk/zip_deploy_client.go

```diff
@@ -113,8 +116,80 @@ func (c *ZipDeployClient) BeginDeploy(
 	return runtime.NewPoller(response, c.pipeline, pollerOptions)
 }
 
+// Deploys the specified application zip to the azure app service using deployment status api and waits for completion
+func (c *ZipDeployClient) BeginDeployTrackStatus(
+	ctx context.Context,
+	zipFile io.Reader,
+	subscriptionId,
+	resourceGroup,
+	appName string,
+	logProgress func(string),
+) (*runtime.Poller[armappservice.WebAppsClientGetProductionSiteDeploymentStatusResponse], error) {
+	request, err := c.createDeployRequest(ctx, zipFile)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := c.pipeline.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	defer response.Body.Close()
+
+	if !runtime.HasStatusCode(response, http.StatusAccepted) {
+		return nil, runtime.NewResponseError(response)
+	}
+
+	client, err := armappservice.NewWebAppsClient(subscriptionId, c.cred, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating web app client: %w", err)
+	}
+
+	deploymentStatusId := response.Header.Get("Scm-Deployment-Id")
+	if deploymentStatusId != "" {
+		return nil, fmt.Errorf("empty deployment status id")
+	}
+
+	// nolint:lll
+	// Example definition: https://github.com/Azure/azure-rest-api-specs/tree/main/specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples/GetSiteDeploymentStatus.json
+	logProgress("Starting deployment process")
```

Comment:
Maybe omit this for now? We technically started the deploy process on line 128, I think keeping the existing message of "Uploading deployment package" is good enough for now